presto-main/src/main/java/com/facebook/presto/operator/aggregation/reservoirsample/UnweightedReservoirSample.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.reservoirsample;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class UnweightedReservoirSample
```

Comment:
@atavory Ami, I'd like to see at least one usage of these data structures. They store values in `List<Object>` regardless of the actual data type. This is a lot more expensive than necessary if underlying data type is integer, long or boolean. For these data types, using primitive arrays (int[], long[], boolean[]) is way more efficient and GC-friendly. How many objects do you expect to store? Do we need to copy these list across machines?

Reply:
@mbasmanova 

Thanks for comment.

What is the alternative? I could duplicate the code for each primitive type and use primitive arrays, but, AFAICT about Java (in which I'm not an expert), there is no way to do this generically otherwise.

As to your other questions: IDK how many objects will be asked to be stored (it will be specified by people calling the UDF, much like `numeric_histogram`). The lists will be transferred across machines (I haven't yet written the serialization code).

Reply:
@atavory 

> What is the alternative? I could duplicate the code for each primitive type and use primitive arrays

It may be necessary to write separate implementations for each type.

> IDK how many objects will be asked to be stored (it will be specified by people calling the UDF, much like numeric_histogram). The lists will be transferred across machines (I haven't yet written the serialization code).

This is important because we have a limit of 16MB for a chunk of data being transferred. Hence, compact serialization might be important.

Let's start with a specific use case (a UDF) and figure all this out.

Reply:
@mbasmanova Still adding unittests, but addressed your comment above, I hope. Thanks!